### PR TITLE
[CFM-1078] Hide articles when filtering projects on the activity stream

### PR DESCRIPTION
**Pull request by guytepper**
Currently, articles are showing up when filtering projects on the activity stream. This PR fixes it.

**JIRA**: https://issuetracker.amplexor.com/jira/browse/CFM-1078

### CODE REVIEW

**Comment by ordavidil** [comment]
@guytepper please check this one and fix the PR:
https://github.com/capacity4dev/capacity4more/pull/1184/files#diff-9622240d6fa26d02595d805da0ac0c14R1228

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -1236,15 +1234,16 @@ function c4m_message_load_homepage_activity_stream() {
       case 'groups':
         // Gets all the group IDs that the user belongs to.
         $groups = c4m_og_get_user_group_nodes_by_bundle('group');
+        $hide_articles = 1;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: remove this

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -1236,15 +1234,16 @@ function c4m_message_load_homepage_activity_stream() {
...
         break;
 
       case 'projects':
         // Gets all the project IDs that the user belongs to.
         $groups = c4m_og_get_user_group_nodes_by_bundle('project');
+        $hide_articles = 1;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: remove this

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 3]
```diff
@@ -1236,15 +1234,16 @@ function c4m_message_load_homepage_activity_stream() {
...
         break;
 
       case 'interests':
-        $hide_articles = 0;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]:
```php
// add a comment.
$hide_articles = FALSE;
```

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 4]
```diff
@@ -1225,9 +1225,7 @@ function c4m_message_load_homepage_activity_stream() {
     : '';
   $groups = array();
   $interests = array();
-  $hide_articles = $filter == 'groups'
-    ? 1
-    : 0;
+  $hide_articles = 0;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]:
```php
$hide_articles = TRUE;
```

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module (hunk 2 above)
Comment (guytepper) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: why? the default value is false

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: look at the other comments as well ...

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module (hunk 4 above)
Comment (guytepper) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: It will hide the articles on the default view ("Show All")

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: set it to false then

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module (hunk 1 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: set it to true

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module (hunk 2 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: set it to true

File: project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module (hunk 3 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: ignore this

**Comment by ordavidil** [comment]
@guytepper 
Can you add a behat test for that logic ?

**Comment by guytepper** [comment]
@ordavidil done

File: project/profiles/capacity4more/behat/features/homepage.feature [hunk 5]
```diff
@@ -126,9 +126,16 @@ Feature: Test homepage content and blocks
     Given I am logged in as user "isaacnewton"
     When  I visit the site homepage
     And   I select the radio button "My groups" with the id "edit-homepage-filter-groups"
-    Then  I should wait not to see "Nobel Prize" in the "div.activity-stream" element
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/homepage.feature]: @guytepper 
why did you change this test ?

Comment (guytepper) [inline project/profiles/capacity4more/behat/features/homepage.feature]: 1. The test is checking for article activities visibility, not for the group activities.
2. The article activities text is different then what was written in the test:

screen shot 2016-08-18 at 1 55 27 pm

image

**Comment by guytepper** [comment]
@ordavidil, the following test is passing on my local machine.. any idea why travis fails? 
screen shot 2016-08-22 at 2 03 28 pm

**Comment by guytepper** [comment]
@ordavidil ?

**Comment by ordavidil** [comment]
@guytepper 
I've re-rolled, please check again on your local machine if the test still fails on travis.

**Comment by guytepper** [comment]
@ordavidil still fails..
